### PR TITLE
fix(frontend): simplify password matching logic

### PR DESCRIPTION
**Pull request by belikedeep**
## Pull Request

---

### 📄 Summary
Fixes a bug where users could bypass password matching validation during signup by pressing `Enter` rapidly. 

https://github.com/user-attachments/assets/97dc0ff5-8d65-4435-8a54-cced78f050e0

#### Issues closed by this PR
N/A

---

### ✅ Change Type
_Select all that apply_

- [ ] ✨ Feature
- [x] 🐛 Bug fix
- [x] ♻️ Refactor
- [ ] 🛠️ Infra / Tooling
- [ ] 🧪 Test-only

---

#### Root Cause
The validation relied on `onBlur`. If a user pressed `Enter` while still typing in the confirm password field, the blur event never fired, and the mismatched passwords were submitted.

#### Fix Strategy
Added strict manual validation directly inside `handleSubmit` to block mismatched passwords, and simplified the UI error state logic using `useWatch`.

---

- Tests added/updated: No
- Manual verification: Yes
- Edge cases covered: -

---

- Blast radius: SignUp Page
- Potential regressions: -
- Rollback plan: Revert this commit

---

| Field | Value |
|------|-------|
| Deployment Type | Cloud / OSS / Enterprise |
| Change Type | Bug Fix |
| Description | Fixed a race-condition issue on UI allowing the first user to be created even when passwords does not match. |

---

### 📋 Checklist
- [x] Tests added or explicitly not required
- [x] Manually tested
- [x] Breaking changes documented
- [x] Backward compatibility considered

---

## 👀 Notes for Reviewers

---

> [!NOTE]
> **Medium Risk**
> Medium risk because it changes signup form validation/submit flow and could inadvertently block valid signups or alter error display behavior, though it is confined to frontend UI logic.
> 
> **Overview**
> Prevents signup submission when `password` and `confirmPassword` don’t match by adding an explicit check inside `handleSubmit`, closing a path where rapid `Enter` presses could bypass blur-based validation.
> 
> Simplifies password-mismatch UI handling by deriving mismatch state from `useWatch` values and showing inline field errors after the confirm-password field is touched, removing the separate mismatch callout and related blur/value-change handlers.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 25e7420f900ff826c24651cb78a529a22d7d2156. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: frontend/src/pages/SignUp/SignUp.tsx [hunk 1]
```diff
@@ -66,6 +65,9 @@ function SignUp(): JSX.Element {
 		(async (): Promise<void> => {
 			try {
 				const values = form.getFieldsValue();
+				if (values.password !== values.confirmPassword) {
+					return;
+				}
```
[the remark this answers is not in the document]
Comment (belikedeep) [reply frontend/src/pages/SignUp/SignUp.tsx]: fixed

File: frontend/src/pages/SignUp/SignUp.tsx [hunk 2]
```diff
@@ -185,6 +156,12 @@ function SignUp(): JSX.Element {
 								<FormContainer.Item
 									name="confirmPassword"
 									validateTrigger="onBlur"
+									validateStatus={showPasswordMismatchError ? 'error' : ''}
+									help={
+										showPasswordMismatchError
+											? "Passwords don't match. Please try again."
+											: ''
+									}
```
[the remark this answers is not in the document]
Comment (belikedeep) [reply frontend/src/pages/SignUp/SignUp.tsx:161]: fixed

**Review by H4ad — APPROVED** [review]
Thanks for the PR, this is great catch!

**Review by AshwinBhatkal — APPROVED** [review]
